engine/resource/src/dmsdk/resource/resource.h

```diff
@@ -111,6 +111,8 @@ namespace dmResource
         const void* m_Buffer;
         /// Size of data buffer
         uint32_t m_BufferSize;
+        /// Pointer holding a precreated message
+        const void* m_Message;
```

Comment:
Same as the recreate params. The m_Buffer ptr is expected to hold an actual DDF message, but we need to support passing in custom data to the resource pipeline as well (unless we want to build DDF messages in runtime, which seems really tricky..)